otter/test/test_controller.py

```diff
@@ -46,9 +51,63 @@
 from otter.util.retry import (
     Retry, ShouldDelayAndRetry, exponential_backoff_interval, retry_times)
 from otter.util.timestamp import MIN
+from otter.util.zk import DeleteNode
 from otter.worker_intents import EvictServerFromScalingGroup
 
 
+class PauseGroupTests(SynchronousTestCase):
+    """
+    Tests for `conv_pause_group_eff`
+    """
+
+    def setUp(self):
+        self.group = util_mock_group(sample_group_state(), "tid", "gid")
+        self.log = mock_log()
+
+    def test_conv_pause_group_eff(self):
+        """
+        `conv_pause_group_eff` returns effect that modifies group state paused
+        and deletes divergent flag with bound log context
+        """
+        eff = controller.conv_pause_group_eff(self.group, "transid")
+        seq = [
+            (BoundFields(mock.ANY, dict(transaction_id="transid",
+                                        tenant_id="tid",
+                                        scaling_group_id="gid")),
+             nested_sequence([
+                 nested_parallel([
+                     (ModifyGroupStatePaused(self.group, True), noop),
+                     (DeleteNode(path="/groups/divergent/tid_gid", version=-1),
+                      noop),
+                     (Log("mark-clean-success", {}), noop)
+                 ])
+             ]))
+        ]
+        self.assertEqual(perform_sequence(seq, eff), None)
+
+    @mock.patch("otter.controller.conv_pause_group_eff",
```

Comment:
It seems the mock here is buying us very little, since I think it should be pretty easy to check the real intents used. You could factor these tests so that they share an intent-sequence.